src/aggregate.js

```diff
@@ -38,31 +42,24 @@ const config = require('./config');
  */
 module.exports = async function aggregate(collection, params) {
   params = _.defaults(await sanitizeParams(collection, params), { aggregation: [] });
-  const cursorQuery = generateCursorQuery(params);
-  const $sort = generateSort(params);
 
-  let index = _.findIndex(params.aggregation, (step) => !_.isEmpty(step.$match));
+  const $match = generateCursorQuery(params);
+  const $sort = generateSort(params);
+  const $limit = params.limit + 1;
 
-  if (index < 0) {
-    params.aggregation.unshift({ $match: cursorQuery });
-    index = 0;
+  let sortSteps;
+  if (params.sortCaseInsensitive) {
+    sortSteps = [{ addFields: { __lc: { $toLower: '$' + params.paginatedField } } }, { $sort }];
   } else {
-    const matchStep = params.aggregation[index];
-
-    params.aggregation[index] = {
-      $match: {
-        $and: [cursorQuery, matchStep.$match],
-      },
-    };
+    sortSteps = [{ $sort }];
   }
 
-  params.aggregation.splice(index + 1, 0, { $sort });
-  params.aggregation.splice(index + 2, 0, { $limit: params.limit + 1 });
+  const aggregation = params.aggregation.concat([{ $match }, ...sortSteps, { $limit }]);
```

Comment:
I don't know if that is a good approach because it would lock those 3 operations at the end.
I.e. maybe we want to use a `$lookup` after the sort & limit phase.

What do you think of giving the flexibility for the user to choose where to add those operations?
Something like that:

The user can _optionally_ supply the `aggregation` array with 3 special values:

```
paging.aggregate(collection, {
   aggregation: [
      { $match: /* custom match */ },
      '$paginationMatch',
      '$paginationSort',
      '$paginationLimit', 
      { $lookup: /* custom lookup */ }, 
      { $project: /* custom project */ }
    ]
}
```

Reply:
Following my previous comment and thinking about it a little, I believe it is better not to allow the caller this flexibility: the only reason to do the pagination before the end is performance, but it opens the door to all sorts of mistakes which result from changing the collection after the pagination (re-sorting, filtering, unrolling, grouping... even projecting if the pagination field is redacted out... all would result in functional errors).

We will think about this if/when we actually hit those performance issues.